californium-core/src/main/java/org/eclipse/californium/core/network/NullMessageIdTracker.java

```diff
@@ -0,0 +1,58 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Bosch Software Innovations GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Bosch Software Innovations - initial creation
+ *                                 (derived from MessageIdTracker)
+ ******************************************************************************/
+package org.eclipse.californium.core.network;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * A helper for keeping track of message IDs.
+ * <p>
+ * According to the
+ * <a href="https://tools.ietf.org/html/rfc7252#section-4.4">CoAP spec</a>
+ * 
+ * <pre>
+ * The same Message ID MUST NOT be reused (in communicating with the
+   same endpoint) within the EXCHANGE_LIFETIME (Section 4.8.2).
+ * </pre>
+ * 
+ * This implementation just increments the MIDs and ignores the RFC7252, 4.8.2.
+ */
+public class NullMessageIdTracker implements MessageIdTracker {
+
+	/**
+	 * Current MID.
+	 */
+	private AtomicInteger currentMID = new AtomicInteger();
+
+	/**
+	 * Creates a new tracker based on configuration values.
+	 * 
+	 * @param initialMid initial MID.
+	 */
+	public NullMessageIdTracker(int initialMid) {
+		currentMID.set(initialMid);
+	}
+
+	/**
+	 * Gets the next message ID.
+	 * 
+	 * @return a message ID.
+	 */
+	public int getNextMessageId() {
+		return currentMID.getAndIncrement() % TOTAL_NO_OF_MIDS;
```

Comment:
Hmm, aren't the numbers correct when interpreting them as unsigned 16bit integers, i.e. 

the binary representation of -65535 as a signed 32bit integer is

11111111 11111111 00000000 00000001, isn't it? So, if you just use the 16 least significant bits and interpret them as an unsigned int, then you get
00000000 00000001 = 1
right?